integration_tests/pages/homePage.ts

```diff
@@ -0,0 +1,17 @@
+import { expect, type Locator, type Page } from '@playwright/test'
+import AbstractPage from './abstractPage'
+
+export default class HomePage extends AbstractPage {
+  readonly header: Locator
+
+  private constructor(page: Page) {
+    super(page)
+    this.header = page.locator('h1', { hasText: 'This site is under construction...' })
+  }
+
+  static async verifyOnPage(page: Page): Promise<HomePage> {
+    const homePage = new HomePage(page)
+    await expect(homePage.header).toBeVisible()
+    return homePage
+  }
+}
```

Comment:
we could have had something generic on `AbstractPage` like we did in cypress but it's a bit unwieldy:
```ts
static verifyOnPage<T>(constructor: new (page: Page) => T, page: Page): T {
    const result = new constructor(page)
    expect(result.locationVerifier).isVisible()
    return result
}
...
 class MyPage {
   constructor(page:Page) {
     super(page, page.locator(...))
   }
 }
...
const myPage = AbstractPage.verifyOnPage(MyPage, page)
```